inc/commondbtm.class.php

```diff
@@ -36,64 +36,191 @@
+   /**
+    * Data fields of the Item.
+    *
+    * @var mixed[]
+    */
+   public $fields = [];
+
+   /**
+    * Flag to determine whether or not changes must be logged into history.
+    *
+    * @var boolean
+    *
+    * @TODO This variable should not be updated and could be converted as a constant (protected if min PHP >= 7.1).
+    */
+   public $dohistory = false;
+
+   /**
+    * List of fields that must not be taken into account when logging history or computating last
+    * modification date.
+    *
+    * @var string[]
+    *
+    * @TODO This variable should not be updated and could be converted as a constant (protected if min PHP >= 7.1).
+    */
+   public $history_blacklist = [];
+
+   /**
+    * Flag to determine whether or not automatic messages must be generated on actions.
+    *
+    * @var boolean
+    *
+    * @TODO This variable should not be updated and could be converted as a constant (protected if min PHP >= 7.1).
+    */
+   public $auto_message_on_action = true;
+
+   /**
+    * Flag to determine whether or not a link to item form can be automatically generated via
+    * self::getLink() method.
+    *
+    * @var boolean
+    *
+    * @TODO This variable should not be updated and could be converted as a constant (protected if min PHP >= 7.1).
+    */
+   public $no_form_page = false;
+
+   /**
+    * Flag to determine whether or not table name of item can be automatically generated via
+    * self::getTable() method.
+    *
+    * @var boolean
+    *
+    * @TODO This variable should not be updated and could be converted as a constant (protected if min PHP >= 7.1).
+    */
+   static protected $notable = false;
+
+   /**
+    * List of fields that must not be taken into account for dictionnary processing.
+    *
+    * @var string[]
+    *
+    * @TODO This variable should not be updated and could be converted as a constant (protected if min PHP >= 7.1 ? it is used in ocsinventoryng).
+    */
+   public $additional_fields_for_dictionnary = [];
 
-   /// Set true to desactivate link generation because form page do not permit show/edit item
-   public $no_form_page                        = false;
+   /**
+    * List of linked item types on which entities informations should be forwarded on update.
+    *
+    * @var string[]
+    *
+    * @TODO This variable should not be updated and could be converted as a constant (protected if min PHP >= 7.1).
+    */
+   static protected $forward_entity_to = [];
+
+   /**
+    * Foreign key field cache : set dynamically calling getForeignKeyField
+    *
+    * @TODO Remove this variable as it is not used ?
+    */
+   protected $fkfield = "";
 
-   /// Set true to desactivate auto compute table name
-   static protected $notable                   = false;
+   /**
+    * Search option of item. Initialized on first call to `self::getOptions()` and used as cache.
+    *
+    * @var array
+    *
+    * @TODO Should be static if used as singleton.
```

Comment:
If this is used as cache this should be dropped in favor of real cache.

Reply:
TODO Updated.